### PR TITLE
Use boot3 notation when restarting a batch job

### PR DESCRIPTION
**Pull request by cppwfs**
Following updates made:
1. Now supports Long type for boot3 app job parameters
2. Supports default format for boot 3 job parameters 
3. Added tests for both Boot2 and boot 3 job parameters.

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java [hunk 1]
```diff
@@ -252,8 +253,10 @@ public void restartJobExecution(long jobExecutionId, String schemaTarget) throws
 	 * @return deduped list of arguments that contains the original arguments and any
 	 * identifying job parameters not in the original task execution arguments.
 	 */
-	private List<String> restartExecutionArgs(List<String> taskExecutionArgs, JobParameters jobParameters) {
+	private List<String> restartExecutionArgs(List<String> taskExecutionArgs, JobParameters jobParameters, String schemaTarget) {
 		List<String> result = new ArrayList<>(taskExecutionArgs);
+		final String BOOT3_VERSION = SchemaVersionTarget.createDefault(AppBootSchemaVersion.BOOT3).getName();
```
Comment (onobc) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java:258]: Please remove final. Also, upper case signifies a static constant.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java [hunk 2]
```diff
@@ -265,9 +268,27 @@ private List<String> restartExecutionArgs(List<String> taskExecutionArgs, JobPar
 					}
 				}
 				if (!existsFlag) {
-					result.add(String.format("%s(%s)=%s", key,
+					String param;
+					if (BOOT3_VERSION.equals(schemaTarget)) {
+						if (JobParameter.ParameterType.LONG.equals(jobParametersMap.get(key).getType())) {
```
Comment (onobc) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java:273]: I assume by this time that `jobParametersMap.get(key)` is always non-null?

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java [hunk 3]
```diff
@@ -265,9 +268,27 @@ private List<String> restartExecutionArgs(List<String> taskExecutionArgs, JobPar
...
+							type = "java.lang.Long";
```
Comment (onobc) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java:274]: Might be less error-prone to use `Long.class.getSimpleName()`

**Review by onobc — CHANGES_REQUESTED** [review]
LGMT @cppwfs - a few minor comments to address. Thanks for the updates.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java (hunk 2 above)
Comment (cppwfs) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java:273]: Yes.  At line jobParametersMap is a non modifiable map and would have thrown an NPE above this.   Good question

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java (hunk 3 above)
Comment (cppwfs) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java:274]: Will do.   FYI it is : `Long.class.getCanonicalName()`

Comment (onobc) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/service/impl/DefaultTaskJobService.java]: In this case, `getCanonicalName() == getSimpleName()` as it is not an array, inner class, nested, etc..

**Review by onobc — APPROVED** [review]

**Pull request merged**